src/sys/ptrace/linux.rs

```diff
@@ -152,6 +157,80 @@ libc_enum! {
     }
 }
 
+#[cfg(all(target_os = "linux", target_env = "gnu"))]
+#[cfg_attr(docsrs, doc(cfg(all())))]
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
+pub struct SyscallInfo {
```

Comment:
Is it possible (and convenient) to reuse the `ptrace_syscall_info` type from libc, like this:

```rs
#[repr(transparent)]
#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
pub struct SyscallInfo(ptrace_syscall_info);
```

If so, then we [should switch to it](https://github.com/nix-rust/nix/blob/master/CONVENTIONS.md#libc-constants-functions-and-structs)

Reply:
libc ptrace_syscall_info has a union, which is not safe to use. So the safe SyscallInfo here is probably preferred.